### PR TITLE
Close #7144: Add CSS class indicating its domain for each desc node

### LINKED ISSUE #7144: Add CSS class indicating its domain for each desc node
**Is your feature request related to a problem? Please describe.**
As commented at https://github.com/sphinx-doc/sphinx/pull/6990#issuecomment-571194243, it would be better to decorate (change a style) every element of object descriptions. To do that, it would be useful to give a domain-named CSS class for each desc element to select specific domain.

**Describe the solution you'd like**
Give a domain-named CSS class for each desc element.
```
<desc classes=['py', 'function']>
  <desc_signature>...</desc_signature>
  <desc_content>...</desc_content>
</desc>
```
This structure enables themes to give a style with a CSS selector like `.py.function > .attribute-name`.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Feature

### Purpose
- refs: #7144

### CODE REVIEW

File: sphinx/directives/__init__.py [hunk 1]
```diff
@@ -160,6 +160,8 @@ def run(self) -> List[Node]:
         # 'desctype' is a backwards compatible attribute
         node['objtype'] = node['desctype'] = self.objtype
         node['noindex'] = noindex = ('noindex' in self.options)
+        if self.domain:
+            node['classes'].append(self.domain)
```
Comment (tk0miya) [inline sphinx/directives/__init__.py:164]: Additional testcases are needed.

File: tests/test_build_html.py [hunk 2]
```diff
@@ -176,8 +176,8 @@ def test_html4_output(app, status, warning):
          r'-|      |-'),
     ],
     'autodoc.html': [
-        (".//dt[@id='autodoc_target.Class']", ''),
-        (".//dt[@id='autodoc_target.function']/em", r'\*\*kwds'),
+        (".//dl[@class='py class']/dt[@id='autodoc_target.Class']", ''),
+        (".//dl[@class='py function']/dt[@id='autodoc_target.function']/em", r'\*\*kwds'),
```
Comment (tk0miya) [inline tests/test_build_html.py:180]: After this change, `py` class will be added into `<dl>` element. So users can give a custom style only for python functions via this class and CSS selector.

**Review by shimizukawa — APPROVED** [review]
LGTM!

**Comment by tk0miya** [comment]
Thank you for reviewing!

**Pull request merged**